doc/userguide/rules/rule-types.rst

```diff
@@ -0,0 +1,930 @@
+.. role:: example-rule-action
+.. role:: example-rule-header
+.. role:: example-rule-options
+.. role:: example-rule-emphasis
+
+Rule Types and Categorization
+=============================
+
+Once parsed, Suricata rules are categorized for performance and further
+processing (as different rule types will be handled by specific engine modules).
+The signature types are defined in `src/detect.h
+<https://github.com/OISF/suricata/blob/master/src/detect.h>`_:
+
+.. literalinclude:: ../../../src/detect.h
+    :caption: src/detect.h
+    :language: c
+    :start-after: // rule types documentation tag start: SignatureType
+    :end-before: // rule types documentation tag end: SignatureType
+
+In more human readable terms:
+
+.. list-table:: Suricata Rule Types, and their Engine Analysis Term
+    :header-rows: 1
```

Comment:
I know it's not ideal as is, but I wanted to avoid having even more columns, as some of the cels in the other column are quite busy with text and Read the Docs doesn't offer a lot of flexibility for adjusting column width...